### PR TITLE
feature(ci): protobuf compatibility check (pt 2)

### PR DESCRIPTION
**Pull request by dsmith3197**
Adds a Github workflow that does a protobuf compatibility check. Verified that the workflow as expected.

### CODE REVIEW

File: lib/vector-core/proto/event.proto [hunk 1]
```diff
@@ -95,7 +107,7 @@ message Metric {
   }
   string namespace = 11;
   uint32 interval_ms = 18;
-  Value metadata = 19;
+  Metadata metadata = 19;
```
Comment (dsmith3197) [inline lib/vector-core/proto/event.proto:110]: I've added a breaking proto change as an example to test the workflow. I'll revert these changes before merging.

File: .github/workflows/protobuf.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Protobuf Compatibility
+
+on:
+  pull_request:
```
Comment (neuronull) [inline .github/workflows/protobuf.yml:4]: I'm wondering if we should do this on PRs. I'm guessing the runtime isn't very long? But given the low frequency with which we make edits to this area, I would think just having it in the merge queue would be plenty sufficient.

File: .github/workflows/protobuf.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    paths:
+        - "proto/**"
+        - "lib/vector-core/proto/**"
+  merge_group:
+    types: [checks_requested]
+
+concurrency:
+    # `github.event.number` exists for pull requests, otherwise fall back to SHA for merge queue
+    group: ${{ github.workflow }}-${{ github.event.number || github.event.merge_group.head_sha }}
```
Comment (neuronull) [inline .github/workflows/protobuf.yml:13]: just noting that if we decide not to do for PRs, can remove the `event.number`

File: .github/workflows/protobuf.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      # Run `git checkout`
+      - uses: actions/checkout@v3
+      # Install the `buf` CLI
+      - uses: bufbuild/buf-setup-action@v1.26.1
+      # Perform breaking change detection against the `master` branch
+      - uses: bufbuild/buf-breaking-action@v1.1.3
+        with:
+          against: "https://github.com/vectordotdev/vector.git#branch=dougsmith/buf"
```
Comment (neuronull) [inline .github/workflows/protobuf.yml:27]: just flagging to change to the master branch before final merge

**Review by pront — APPROVED** [review]
Thanks for adding this workflow! I agree it is valuable to have this check run in PRs.

**Pull request merged**